### DISCUSSION TITLE
Can't connect to local WS Server Versa3 with SDK5

### DISCUSSION THREAD

**Discussion by bizzclaw** [200Tigersbloxed/FitbitHRtoWS#9] [Q&A]
Hello! I'm currently trying to set this up using SDK5 and I'm having trouble connecting the app to a local server.
- I set up a custom password and made sure they were the same on the config.json and the app config. 
- I got my local IP as 10.0.0.58 so I set the URi to ws://10.0.0.85:8080 on the app settings
- I forwarded the port range of 8000-8080
- I unblocked those ports in the windows firewall on both UDP and TCP
- The watch is connected to the same wifi network as the PC and I've even tried restarting the watch after changing the settings
- Connection status on the Fitbit is just "Please Wait" and the Heart Rate is "---"

I'm a former web developer so I know a little bit about websockets. I also know how much of a pain they are so I appreciate whatever support can be given.

**Comment by 200Tigersbloxed** [comment]
Thanks for the information! Just want to do some basic troubleshooting steps as Fitbit can be very touchy to work with.

*Fitbit*
1) Verify your watch is connected to your phone via. bluetooth at all times.
2) Make sure you're leaving the Fitbit app on your phone open at all times. (The app is the bridge that connects the watch to the internet!)
3) Verify your app can connect and see your Versa 3 (sync your watch)

*Server*
1) Try going to http://10.0.0.58:8000 on another device
  * The Websocket will show as disconnected no matter what, just making sure ports are forwwarded
  * also you only have to have 8080 and 8000 open on TCP

After these steps, if you still experience issues, let me know!

**Reply by 200Tigersbloxed** [reply]
For some more info,

When the watch says "Please Wait..." that means it's trying to make a connection to the fitbit app's bridge (companion API). If it can't connect to a server, it'll say Not Connected.

You can read about Fitnit's Companion API WebSocket interface [here](https://dev.fitbit.com/build/reference/companion-api/websocket/)

**Reply by bizzclaw** [reply]
I can access http://10.0.0.58:8000 on my phone on wifi and everything shows up.
![Screenshot_20211208-124720_Brave](https://user-images.githubusercontent.com/3189582/145284806-c0948d9f-b854-4291-a50c-16d926fd4aff.jpg)

I have the fitbit app running showing the Versa 3 main page. I'm connected to wifi and the watch is connected but still no connection.

Question: Do I forward the ports for the desktop Local IP, or also the local IP for my phone?

**Reply by bizzclaw** [reply]
Update: I'm getting this issue on and off where on the Fitbit mobile app, it will say "Oops, looks like there's no connection" People are saying it's related to having an ad-blocking app, but I don't have anything like that and it only happens whenever I'm on wifi, if I switch to mobile data the app works fine, so, for whatever reason, the mobile app for Fitbit is having trouble connecting to my wifi through my phone, even though other things on my phone can use wifi just fine. Very Strange, but this might be something out of your hands to fix and an issue on my end with my phone or the FitBit app.
![Screenshot_20211208-124224_Fitbit](https://user-images.githubusercontent.com/3189582/145291089-f85deefb-ff76-4d66-8a38-a09c60377c67.jpg)

**Reply by bizzclaw** [reply]
One more thing, I can confirm that the public server works.

**Comment by 200Tigersbloxed** [comment]
From someone else who got this working after it not working,

> i have good news about this weird error [Local Server not Working] people are getting
its just fitbit being a jerk
i got it running on the server and public now so it can do both nothing change except several reinstalls of the Fitbit app and fac resets of the watch till it worked.

Other than that, I have no idea what the issue would be.

Also, if your app says it isn't connected to the internet, then it won't work properly, even if you're just using a local server.
